### PR TITLE
Proper handling when using keywords as the order option

### LINKED ISSUE #2: Problem when using MySQL reserved words
If you use a reserved word as the `order` field the the generated SQL does not get properly escaped when `quoteIdentifiers` is set to true.

Not sure if this is a bug with Cake or not.

__Example__
````php
$this->addBehavior('Sequence.Sequence', [
    'order' => 'order',
    'start' => 0,
    'scope' => 'course_id'
]);
```
Then the following SQL is created:
 ```sql
UPDATE course_modules SET `order` = (order - 1) WHERE (`order` > :c0 AND `course_id` = :c1)
```
You can see the order is not getting escaped. It's caused by these lines 
```php
$this->_table->query()->newExpr()->add("$orderField + 1")
```

### PR DESCRIPTION
**Pull request by veganista**
Should fix #2

### CODE REVIEW

File: src/Model/Behavior/SequenceBehavior.php [hunk 1]
```diff
@@ -363,4 +364,21 @@ protected function _sync($fields, $conditions, $scope = null)
...
         return $this->_table->updateAll($fields, $conditions);
     }
+
+    /**
+     * Returns the update expression for the order field
+     *
+     * @param string $direction Whether to increment or decrement the field
+     * @param string $amount Amount to modify the field by
+     */
+    protected function _getOrderUpdateExpression($direction = 'increment', $amount = 1) {
```
Comment (ADmad) [inline src/Model/Behavior/SequenceBehavior.php]: - Method name is a bit too long, just `_getUpdateExpression()` is good enough. 
- No need to use strings `increment`, `decrement` for direction, just directly use `+`, `-`.
- `$amount` param can probably be dropped. I don't think there will be a use case of amount value other than 1.

File: src/Model/Behavior/SequenceBehavior.php [hunk 2]
```diff
@@ -363,4 +364,21 @@ protected function _sync($fields, $conditions, $scope = null)
...
+        $field = $this->_config['order'];
+
+        if ($direction == 'decrement') {
+            $direction = '-';
+        } else {
+            $direction = '+';
+        }
+        return $this->_table->query()->newExpr()->add(new IdentifierExpression($field))->add((string)$amount)->type($direction);
```
Comment (ADmad) [inline src/Model/Behavior/SequenceBehavior.php]: Please format like below for better readability:

```php
return $this->_table->query()->newExpr()
    ->add(new IdentifierExpression($field))
    ->add('1')
    ->type($direction);
```

**Comment by ADmad** [comment]
Please fix all [CS errors](https://travis-ci.org/ADmad/cakephp-sequence/jobs/52006653)

**Comment by ADmad** [comment]
Final step, squash commits :)

**Comment by ADmad** [comment]
Thanks @veganista

**Comment by veganista** [comment]
Cheers for your help on the last bit!

**Pull request merged**